### PR TITLE
Add analytics (DAT-17058)

### PR DESCRIPTION
**Pull request by liquibot**
This PR introduces the ability for Liquibase to gather anonymous usage analytics when commands are run. No user-specific data will be gathered. 

For Liquibase OSS users, this functionality will be enabled by default, but can be opted-out of before any data is sent. For Liquibase Pro users, this will disabled by default with the ability to opt-in. 

Only the versions of Liquibase after this PR goes into a future release will have the ability to gather anonymous analytics. Those versions will have this functionality on or off immediately when used for the first time as specified above. There are sections below that instruct how to turn this on or off as desired. 

## Why are we doing this?

Gathering anonymous usage analytics allows us to better understand how Liquibase is actually being used by a very wide, diverse set of millions of users. This data will help our small team help our users better by providing insight into things like:

- What commands are the most used:  if some commands are never used it might mean we need to reconsider that command’s purpose, provide more resources to help increase usage, or even remove the command entirely.
- What commands result in the most failures: knowing this will help us identify gaps in documentation or the product itself.
- What database platforms are the most used with Liquibase: this information will help us identify where we should focus our efforts to maximize benefit for our users.
- What version of Java are our users on: this will help us as we consider updating Java versions in core and many extensions.
- What version of Liquibase are users on: this will help us understand if users are updating regularly or where we might help with updates.
- What extensions of Liquibase are most loaded and used: this will help us understand which extensions are being used and we should focus on.

## What information are we gathering?
If the functionality is enabled and a user is not opted-out, Liquibase will record the following information when a command is run:

- Command-specific data like: command name, if the command succeeded or failed, and exception data if there is a failure
- Liquibase data including version and interface that it’s being run in
- Database platform including version 
- Changelog format and number of changelogs run
- Extension information like which extensions are being used and what version
- Pro feature information like which features are enabled
- Operating system information like type of OS, version, and architecture

To reiterate, we are not gathering any personally identifiable information. **This data is all aggregated and anonymous**.

## Where to view the information we gather

We are working to make analytics events and certain parts of the data that we collect publicly available on the web. 

## Opting Out
  
If you do not want to participate in Liquibase anonymous analytics, there are a number of ways to opt-out at any time. 

Per CLI command, you can add the parameter `--analytics-enabled=false`. This will turn off analytics gathering for that single occurrence of a command (ie. `liquibase status –analytics-enabled=false`).

If you would like to turn off the functionality globally, you can add `liquibase.analytics.enabled: false` to your defaults file or use the environment variable LIQUIBASE_ANALYTICS_ENABLED: false.

## Opting In

If you are a Pro user or someone that previously disabled Liquibase anonymous analytics, there are a number of ways to opt-in. 

Per CLI command, you can add the parameter `--analytics-enabled=true`. This will turn on analytics gathering for that single occurrence of a command (ie. `liquibase status --analytics-enabled=true`).

If you would like to turn on the functionality globally, you can add `liquibase.analytics.enabled: true` to your defaults file or use the environment variable LIQUIBASE_ANALYTICS_ENABLED: true.

## More Information

If you have any questions or feedback please respond to this issue or email us - product@liquibase.com.

### CODE REVIEW

File: liquibase-standard/src/main/java/liquibase/analytics/AnalyticsBatch.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import liquibase.analytics.configuration.AnalyticsConfigurationFactory;
+import lombok.Data;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Data
+public class AnalyticsBatch {
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/analytics/AnalyticsBatch.java:15]: Could you add javadoc here?

File: liquibase-standard/src/main/java/liquibase/analytics/LiquibaseAnalyticsListener.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.introspector.BeanAccess;
+import org.yaml.snakeyaml.nodes.Tag;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+
+@NoArgsConstructor
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/analytics/LiquibaseAnalyticsListener.java:25]: Could you add javadoc here?

File: liquibase-standard/src/main/java/liquibase/analytics/configuration/LiquibaseRemoteAnalyticsConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+
+@Data
+public class LiquibaseRemoteAnalyticsConfiguration implements AnalyticsConfiguration {
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/analytics/configuration/LiquibaseRemoteAnalyticsConfiguration.java:21]: Could you add javadoc here?

**Review by filipelautert — CHANGES_REQUESTED** [review]

**Comment by filipelautert** [comment]
@StevenMassaro I marked 3 points in new classes were some javadoc would be helpful.

**Review by filipelautert — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]

**Comment by sideeffffect** [comment]
In case somebody is interested in turning it off programmatically, here's an easy way
 * https://github.com/sideeffffect/liquibase-doobie/pull/426

**Pull request merged**